src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+
+      const matchesType = type ? row.type.toLowerCase().includes(type.toLowerCase()) : true;
+
+      const matchesCategory = category ? row.category.toLowerCase().includes(category.toLowerCase()) : true;
+
+      // const matchesStatus = status ? item.status.toLowerCase().includes(status.toLowerCase()) : true; //
+
+      const matchesSearch = search
+        ? row.item.toLowerCase().includes(search.toLowerCase()) ||
+        row.type.toLowerCase().includes(search.toLowerCase()) ||
+        row.category.toLowerCase().includes(search.toLowerCase()) ||
+        // item.status.toLowerCase().includes(search.toLowerCase()) ||
+        row.quantity.toString().toLowerCase().includes(search.toLowerCase())
+        : true;
+
+      return matchesType && matchesCategory && matchesSearch;
+    });
+
+    setData(searchFiltered);
+  };
+
+  const supabase = createClient(import.meta.env.VITE_SUPABASE_URL, import.meta.env.VITE_SUPABASE_ANON_KEY); // When we deploy this application or when we create a server, this will have to be changed to server side so that the anon_key can be hidden.
+
+  const fetchData = async () => {
+    const { data: inventory, error } = await supabase
+      .from('inventory')
+      .select(`id, item, type, quantity, category`);
+
+    if (error) {
+      console.error('Error fetching inventory:', error)
+    } else {
+      console.log(inventory);
+      setOriginalData(inventory);
+      setData(inventory)
+    }
+  }
+
+  useEffect(() => {
+    if (type || category || status || search) {
+      handleFilter()
+    } else {
+      fetchData();
+    }
+  }, [type, category, status, search])
+
+
+  return (
+    <Box>
+
+      {/* Add button */}
+      <Box id="add-container" sx={{ display: 'flex', justifyContent: 'end' }}>
+        <Button sx={{ bgcolor: '#F5F5F5', color: 'black' }}><AddIcon fontSize="small" sx={{ color: 'black' }} />Add</Button>
+      </Box>
+
+      {/* Filter Container */}
+      <Box id="filter-container" sx={{ display: 'flex', alignItems: 'center', maxWidth: '90%' }}>
+        <Typography variant="body2">Filters</Typography>
+
+        {/* Type Filter */}
+        <Box sx={{ px: '8px' }} id="type-button-container">
+          <Button sx={{ color: 'black', bgcolor: '#E0E0E0', height: '30px' }} onClick={handleTypeClick}>
+            {type ? (
+              <>{type} <ClearIcon sx={{ fontSize: 'large', ml: '6px' }} onClick={clearTypeFilter} /></>
+            ) : (<><Typography variant="body2">Type</Typography><ExpandMoreIcon sx={{ fontSize: 'large', ml: '6px' }} /></>)}
+          </Button>
+          <Menu
+            open={Boolean(anchorType)}
+            onClose={handleTypeClose}
+            anchorEl={anchorType}
+          >
+            <MenuItem onClick={() => handleMenuTypeClick('Donation')}>Donation</MenuItem>
+            <MenuItem onClick={() => handleMenuTypeClick('Welcome Basket')}>Welcome Basket</MenuItem>
```

Comment:
Same flexibility issue with Catagory menu.

Reply:
I think we could add this functionality into an admin page. I can check with with Bai, her design did not call for adding category/types